wls-gui-wahllokalsystem/src/api/fetch-utils.ts

```diff
@@ -129,21 +130,40 @@ export function wlsResponseHandler(response: Response): Promise<Response> {
 
 export function wlsCatchHandler(response: Response): PromiseLike<never> {
   if (response.status === 204) {
-    throw new ApiError({
+    throw new WLSError({
       level: STATUS_INDICATORS.INFO,
       message: "Es konnten keine Daten gefunden werden",
+      code: response.status.toString(),
     });
   }
   if (response.status === 400) {
     return response.json().then((content) => {
-      if (WLSException.isWLSException(content)) {
-        return Promise.reject(new ApiError({ message: content.message }));
+      if (WLSError.isWLSException(content)) {
```

Comment:
EIn Vorschlag um nicht zweimal reject aufzurufen. Ich denke es macht es auch etwas lesbarer. Das Statement mit dem elvis-Operator könnte man auch in eine Funktion auslagern.

```typescript
      const wlsError = WLSError.isWLSException(content)
        ? new WLSError({
            level: STATUS_INDICATORS.ERROR,
            message: content.message,
            category: content.category,
            code: content.code,
            service: content.service,
          })
        : new WLSError({
            message: "Error: Ungültige Anfrage",
            code: response.status.toString(),
          });
      return Promise.reject(wlsError);
```

Reply:
siehe [ca5ad70](https://github.com/it-at-m/Wahllokalsystem/pull/647/commits/ca5ad702ce91de55df1b78dcdd7d2d2a17f8563c) und [98c7949](https://github.com/it-at-m/Wahllokalsystem/pull/647/commits/98c7949539b1328de42507b82ae5a7fc4468eab9)